### PR TITLE
deps: update to @bpmn-io/extract-process-variables@0.7.0

### PR DESCRIPTION
**Pull request by marstamm**
This PR bumps @bpmn-io/extract-process-variables and ensures that we handle asynchronous extractors correctly

related to https://github.com/bpmn-io/extract-process-variables/pull/24
related to https://github.com/bpmn-io/internal-docs/issues/680

### CODE REVIEW

File: src/provider/camunda-platform/properties/ProcessVariablesProps.js [hunk 1]
```diff
@@ -24,31 +24,36 @@ import {
...
+    const withName = populateElementNames(sortByName(rawVariables));
+
+    setVariables(withName);
+  }, [ element ]);
 
-  // (1) fetch available process variables for given scope
-  const variables = getVariablesForScope(scope, rootElement);
 
   if (!variables.length) {
     return null;
```
Comment (nikku) [inline src/provider/camunda-platform/properties/ProcessVariablesProps.js:53]: Does this still work in the async usage case?

Comment (nikku) [reply src/provider/camunda-platform/properties/ProcessVariablesProps.js:53]: I think I understand now, but just to confirm: 

* We do a shallow test to understand if an element may have process variables. If so, then we return a component to render the variables.
* That component renders something _only_ when `variables` exist

Does it make sense to render an empty state instead, in the unlikely case that no process variables exists?

File: test/TestHelper.js [hunk 2]
```diff
@@ -215,6 +215,27 @@ export async function setEditorValue(editor, value) {
   await act(() => {});
 }
 
+
+export async function expectEventually(fn) {
```
Comment (nikku) [inline test/TestHelper.js:219]: Proposal: Rename this utility to `eventually`.

Makes for cleaner tests, I think:

```javascript
await eventually(() => {
  expect(mockComponent).to.have.been.calledWith(
  sinon.match({
    variables: [
      {
        name: 'OutputVariable_0svilsd',
        info: 'Written in Task_2'
      }
    ]
  });
});
```

File: src/provider/camunda-platform/properties/ProcessVariablesProps.js (hunk 1 above)
Comment (marstamm) [reply src/provider/camunda-platform/properties/ProcessVariablesProps.js:53]: > Does it make sense to render an empty state instead, in the unlikely case that no process variables exists?

We do not have an concept for empty states in Groups, so they always need content. What we could do is use a normal Group and add `No variables available in this scope` as content.

However, this makes it hard to see that no Variables are present from the collapsed state, as we never show a count of 0.  I decided against it and kept the current behavior for now, where the group is not shown at all when no variables are present.

Comment (nikku) [reply src/provider/camunda-platform/properties/ProcessVariablesProps.js:53]: Thanks for elaborating. Makes sense to me.

File: test/TestHelper.js (hunk 2 above)
Comment (philippfromme) [reply test/TestHelper.js]: Isn't this what https://testing-library.com/docs/dom-testing-library/api-async#waitfor does already? Comes with `@testing-library/preact`.

Comment (marstamm) [reply test/TestHelper.js]: I did not know about this one, I'll check it out

Comment (marstamm) [reply test/TestHelper.js]: Works exactly like it, I changed it to use the testing utility instead 👍

**Review by nikku — APPROVED** [review]

**Pull request merged**